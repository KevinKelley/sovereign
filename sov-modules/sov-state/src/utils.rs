#[derive(Debug, PartialEq, Eq)]
pub struct AlignedVec {
    inner: Vec<u8>,
}

impl AlignedVec {
    pub fn new(vector: Vec<u8>) -> Self {
        Self { inner: vector }
    }

    pub fn extend(&mut self, other: &Self) {
        self.inner.extend(&other.inner);
    }

    pub fn into_inner(self) -> Vec<u8> {
        self.inner
    }

    pub fn len(&self) -> usize {
        self.inner.len()
    }

    #[must_use]
    pub fn is_empty(&self) -> bool {
        self.len() == 0
    }
}

impl AsRef<Vec<u8>> for AlignedVec {
    fn as_ref(&self) -> &Vec<u8> {
        &self.inner
    }
}
