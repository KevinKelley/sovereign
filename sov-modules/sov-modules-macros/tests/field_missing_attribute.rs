mod utils;
use sov_modules_macros::ModuleInfo;
use utils::{Context, TestState};

#[derive(ModuleInfo)]
struct TestStruct<C: Context> {
    test_state1: TestState<C::Storage>,

    #[state]
    test_state2: TestState<C::Storage>,
}

fn main() {}
