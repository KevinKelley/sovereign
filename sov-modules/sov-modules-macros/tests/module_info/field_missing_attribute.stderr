error: This field is missing an attribute: add `#[module]` or `#[state]`.
 --> tests/module_info/field_missing_attribute.rs:7:5
  |
7 |     test_state1: StateMap<u32, u32, C::Storage>,
  |     ^^^^^^^^^^^
